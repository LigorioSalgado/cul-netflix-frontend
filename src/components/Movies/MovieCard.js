import React, { Component } from 'react'
import Rating from 'react-rating';

export default class MovieCard extends Component {
  render() {


    return (
        <div className="card">
            <div className="card-image waves-effect waves-block waves-light">
                <a href={`/movie/${this.props.id}`}>
                    <img className="activator" src={this.props.image} alt="movie photo"/>
                </a>
            </div>
            <div className="card-content">
                <span className="card-title activator grey-text text-darken-4">
                    {this.props.title}
                </span>
            
                <Rating
                    initialRating={this.props.rating}
                    readonly
                />
            </div>
      </div>
    )
  }
}
