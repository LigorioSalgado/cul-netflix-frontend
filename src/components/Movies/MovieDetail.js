import React, { Component } from 'react';
import gql from 'graphql-tag';
import  {Query} from 'react-apollo';
import Youtube from 'react-youtube';


const SINGLEMOVIE = gql`

    query SingleMovie($id:ID!){
        movie(id:$id){
            name,
            movie_url,
            sinopsis,
            duration
        }
    }


`

export default class MovieDetail extends Component {

    constructor(props){
        super(props)
        this.state = {
            id:props.match.params.id
        }
    }


  render() {
    const opts = {
        height: '390',
        width: '640',
        playerVars: { 
          autoplay: 1
        }
      };
    return (
      <div className="container">
        <div className="row">
            <div className="col s12">
                <Query query={SINGLEMOVIE} variables={{id:this.state.id}}>
                        {
                            ({loading,data,error}) => {
                                if(error) return <h4>{error}</h4>
                                if(loading) return <h4>Cargando...</h4>
                                return(
                                    <React.Fragment>
                                        <h4>{data.movie.name}</h4>

                                        <Youtube videoId={data.movie.movie_url.split("=")[1]}
                                        opts={opts} 
                                        />

                                    </React.Fragment>
                                )

                            }
                        }
                </Query>
            </div>
        </div>
      </div>
    )
  }
}
