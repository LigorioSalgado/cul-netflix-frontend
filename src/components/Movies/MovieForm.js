import React, { Component } from 'react';
import {Input} from '../../common/Input';
import FileUploader from 'react-firebase-file-uploader';
import firebaseConfig  from '../../firebaseConfig';




export default class MovieForm extends Component {
 constructor(props){
    super(props)
    this.state = {
        name:"",
        genre:"COMEDY",
        director:"",
        cast:[],
        sinopsis:"",
        duration:"",
        realesed_date:"",
        rating:4.5,
        rate:"",
        language:"",
        cover:"",
        movie_url:"",
        progress:0,
        actor:{
            castName:"",
            age:""
        }
    }
 }


 addCast = (e) => {
     e.preventDefault();
    const newCast = {
        name:this.state.actor.castName,
        age:this.state.actor.age
    }
    this.setState(
        {cast:[...this.state.cast,newCast],
         actor:{castName:"",age:""}
        }
    )
 }
 handleCastInput = (e) => {
     const {id,value} =  e.target
     let newCast = {...this.state.actor}
     newCast[id] = value
     this.setState(
         {
             actor:{...newCast}
         }
     )

 }

 CastInput = () => {
    return(
        <React.Fragment>
            
            <div className="col s10">
                <ul>    
                    {this.state.cast.map((actor,index) =>(
                        <li key={index}>{actor.name}</li>
                    ))
                }

                </ul>
            </div>
          
            <div className="col s5 input-field">
                    <Input type="text" 
                        id="castName" name="Name" 
                        value={this.state.actor.castName}
                        setInput={this.handleCastInput}
                        required/>
            </div>
            <div className="col s5 input-field">
                    <Input type="text" 
                        id="age" name="Age" 
                        value={this.state.actor.age}
                        setInput={this.handleCastInput}
                        required/>
            </div>  

            <div className="col s10">
                <a href="" className="waves-effect waves-ligth btn btn-primary"
                onClick={this.addCast}
                >Agregar</a>
            </div>

        </React.Fragment>
        
    )

}



 handleInput =  (e) => {
    const {id,value} =  e.target
     this.setState(
        {[id]:value}
     )
 }

 handleUploadError = (error) => {
    console.log(error)
 }
 
 
 progressFile = (progress) => {
    this.setState({progress})
 }

 handleUploadSucces = (filename) => {
    this.setState({progress:100})
    firebaseConfig.storage()
        .ref("covers")
        .child(filename)
        .getDownloadURL()
        .then(url => this.setState({cover:url}))
        .catch(err => console.log(err) )

 }
 
 


  render() {
    return (
      <div className="container">
        <form>
            <div className="row">
                <div className="col s10 input-field">
                    <Input type="text" 
                        id="name" name="Title" 
                        value={this.state.name}
                        setInput={this.handleInput}
                        required/>
                </div>
                <div className="col s10 ">
                    <label htmlFor="genre">Genre</label>
                    <select  id="genre" value={this.state.genre} className="browser-default" onChange={this.handleInput}>
                        <option value="COMEDY">Comedy</option>
                        <option value="ACTION">Action</option>
                        <option value="DRAMA">Drama</option>
                        <option value="SCIFY">Science Fiction</option>
                        <option value="HORROR">Horror</option>
                    </select>
                </div>

              
                <div className="col s10 input-field">
                    <Input type="text" 
                            id="director" name="Director" 
                            value={this.state.director}
                            setInput={this.handleInput}
                            required/>
                </div>
                {this.CastInput()}

                <div className="col s10 input-field">
                    <textarea  className="materialize-textarea" id="sinopsis" cols="30" rows="10" value={this.state.sinopsis}
                    onChange={this.handleInput}
                    ></textarea>
                    <label htmlFor="sinopsis">Sinopsis</label>
                </div>

                <div className="col s10 input-field">
                    <Input type="text" 
                            id="duration" name="Duration" 
                            value={this.state.duration}
                            setInput={this.handleInput}
                            required/>
                </div>

                <div className="col s10 input-field">
                    <Input type="text" 
                            id="realesed_date" name="Released Date" 
                            value={this.state.realesed_date}
                            setInput={this.handleInput}
                            required/>
                </div>

                <div className="col s10">
                    <label htmlFor="rate">Rate</label>
                    <select  id="rate" value={this.state.rate}  className="browser-default" onChange={this.handleInput} >
                        <option value="A">Clasification A</option>
                        <option value="B">Clasification B</option>
                        <option value="C">Clasification C</option>
                        <option value="B15">Clasification B15</option>
                    </select>
                </div>

                <div className="col s10 input-field">
                    <Input type="text" 
                            id="language" name="Language" 
                            value={this.state.language}
                            setInput={this.handleInput}
                            required/>
                </div>

                <div className="col s10 input-field">
                    <Input type="text" 
                            id="movie_url" name="Movie Url" 
                            value={this.state.movie_url}
                            setInput={this.handleInput}
                            required/>
                </div>

                <div className="col s10 ">
                    <label className="btn btn-primary">
                        <FileUploader
                            hidden
                            accept="image/*"
                            randomizeFilename
                            storageRef={
                                firebaseConfig.storage().ref('covers')
                            }
                            onUploadError={this.handleUploadError}
                            onProgress={this.progressFile}
                            onUploadSuccess={this.handleUploadSuccess}
                        />
                    Agregar Cover
                    </label>
                    <span>Progress: {this.state.progress}%</span>

                </div>
            </div>
        </form>
      </div>
    )
  }
}
