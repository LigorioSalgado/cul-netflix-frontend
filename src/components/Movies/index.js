export  {default as Movies} from './Movies';
export {default as  MovieForm} from './MovieForm';
export {default as MovieDetail} from './MovieDetail';