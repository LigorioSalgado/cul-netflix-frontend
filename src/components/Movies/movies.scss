
.Movies{
    color:#fff;
}

.browser-default {
    background: #000;
    border-color: #000;
    color: #fff;
}

ul > li, span {
    color:#fff;
}

