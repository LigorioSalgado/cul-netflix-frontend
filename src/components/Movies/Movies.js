import React, { Component } from 'react'
import gql from 'graphql-tag';
import {Query}  from 'react-apollo';
import MovieCard from './MovieCard';
import './movies.scss';

const ALLMOVIES = gql`
    query{
        movies{
            _id,
            name,
            cover,
            rating,
            rate
        }
    }

`

export default class Movies extends Component {

     

  render() {
    return (
        <div className="container Movies">
            <div className="row">
                <Query query={ALLMOVIES}>
                    {
                            ({data,error,loading}) => {
                                if(error) return <h4>{"Hubo un Error !! :("}</h4>
                                if(loading) return <h4>Cargando ...</h4>
                                const movies = data.movies.map((movie,index) => (
                                    <div className="col s4" key={index}>
                                       <MovieCard image={movie.cover} 
                                       title={movie.name}
                                       rating={movie.rating}
                                       />
                                    </div>

                                ))   
                                return(
                                    <React.Fragment>
                                        {movies}
                                    </React.Fragment>
                                )
                            }
                    }
                    
                </Query>
            </div>
        </div>
    )
  }
}
