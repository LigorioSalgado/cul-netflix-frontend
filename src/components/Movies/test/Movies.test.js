import React from 'react'
import {shallow} from 'enzyme';

import Movies from "../Movies";


describe("<Movies/>", () => {

    it("Debe ejecutar render correctamente",() => {
        const component =  shallow(<Movies/>)
        expect(component).toMatchSnapshot();
    })


})