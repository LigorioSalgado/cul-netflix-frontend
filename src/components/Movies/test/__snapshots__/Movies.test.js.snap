// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movies/> Debe ejecutar render correctamente 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movies />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <Query
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "movies",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "_id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "cover",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "rating",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "rate",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 136,
                "start": 0,
              },
            }
          }
        >
          [Function]
        </Query>
      </div>,
      "className": "container Movies",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Query
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "movies",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "_id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "cover",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "rating",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "rate",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 136,
                "start": 0,
              },
            }
          }
        >
          [Function]
        </Query>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": undefined,
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "movies",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "_id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "cover",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "rating",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "rate",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 136,
              "start": 0,
            },
          },
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <Query
            query={
              Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": undefined,
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "movies",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "_id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "cover",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "rating",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "rate",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 136,
                  "start": 0,
                },
              }
            }
          >
            [Function]
          </Query>
        </div>,
        "className": "container Movies",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Query
            query={
              Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": undefined,
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "movies",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "_id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "cover",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "rating",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "rate",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 136,
                  "start": 0,
                },
              }
            }
          >
            [Function]
          </Query>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": [Function],
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "movies",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "_id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "cover",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "rating",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "rate",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 136,
                "start": 0,
              },
            },
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
